File: Plugins/Aspose_Email_Java_for_Ruby/lib/asposeemailjava/Outlook/distributionlist.rb
module Asposeemailjava
  module DistributionList
    def initialize()
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/'

        oneOffmembers = Rjb::import('com.aspose.email.MapiDistributionListMemberCollection').new
        oneOffmembers.addItem(Rjb::import('com.aspose.email.MapiDistributionListMember').new("John R. Patrick", "JohnRPatrick@armyspy.com"))
        oneOffmembers.addItem(Rjb::import('com.aspose.email.MapiDistributionListMember').new("Tilly Bates", "TillyBates@armyspy.com"))

        dlist = Rjb::import('com.aspose.email.MapiDistributionList').new("Simple list", oneOffmembers)
        dlist.setBody("Test body")
        dlist.setSubject("Test subject")
        dlist.setMileage("Test mileage")
        dlist.setBilling("Test billing")

        dlist.save(data_dir + "distlist.msg")

        puts "Saved distribution list successfully."
    end
  end
end

File: Examples/src/main/java/com/aspose/email/examples/outlook/msg/DistributionList.java
package com.aspose.email.examples.outlook.msg;

import com.aspose.email.MapiDistributionList;
import com.aspose.email.MapiDistributionListMember;
import com.aspose.email.MapiDistributionListMemberCollection;
import com.aspose.email.examples.Utils;

public class DistributionList {

	public static void main(String[] args) {
		// The path to the resource directory.
		String dataDir = Utils.getSharedDataDir(DistributionList.class) + "outlook/";
				
		MapiDistributionListMemberCollection oneOffmembers = new MapiDistributionListMemberCollection();
		oneOffmembers.addItem(new MapiDistributionListMember("John R. Patrick", "JohnRPatrick@armyspy.com"));
		oneOffmembers.addItem(new MapiDistributionListMember("Tilly Bates", "TillyBates@armyspy.com"));

		MapiDistributionList dlist = new MapiDistributionList("Simple list", oneOffmembers);
		dlist.setBody("Test body");
		dlist.setSubject("Test subject");
		dlist.setMileage("Test mileage");
		dlist.setBilling("Test billing");

		dlist.save(dataDir + "distlist_out.msg");
	}

}
